backend/rest_api_service/main.py

```diff
@@ -61,4 +61,125 @@ async def get_all_participants(response: Response, db: Session = Depends(get_db)
 
     except Exception as e:
         response.status_code = 500
-        return {"message": f"An error occured: {e}"}
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.get("/event/{event_id}", tags=['Event'], status_code=status.HTTP_200_OK)
+async def get_event_by_id(event_id, response: Response, db: Session = Depends(get_db)):
+    try:
+        return await crud.get_event_by_id(event_id=event_id, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.get("/user/{user_id}/events", tags=['Event'], status_code=status.HTTP_200_OK)
+async def get_all_events_of_user_by_user_id(user_id: int, response: Response, db: Session = Depends(get_db)):
+    try:
+        return await crud.get_all_events_of_user_by_user_id(user_id=user_id, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.get("/event/crated_by/{user_id}", tags=['Event'], status_code=status.HTTP_200_OK)
+async def get_all_events_by_creator_id(user_id: int, response: Response, db: Session = Depends(get_db)):
+    try:
+        return await crud.get_all_events_by_creator_id(user_id=user_id, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.get("/event/{event_id}/participants", tags=['Event'], status_code=status.HTTP_200_OK)
+async def get_all_event_participants(event_id: int, response: Response, db: Session = Depends(get_db)):
+    try:
+        return await crud.get_all_event_participants(event_id=event_id, db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.get("/user/{user_id}/events/range_of_time/{start, end}", tags=['Event'], status_code=status.HTTP_200_OK)
+async def get_all_events_by_user_id_ongoing_in_specified_time(user_id: int, start: datetime.datetime,
+                                                              end: datetime.datetime, response: Response,
+                                                              db: Session = Depends(get_db)):
+    try:
+        return await crud.get_all_events_by_user_id_ongoing_in_specified_time(user_id=user_id, start=start, end=end,
+                                                                              db=db)
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occurred: {e}"}
+
+
+@app.post("/event/create/{event}", tags=['Event'], status_code=status.HTTP_201_CREATED)
+async def create_event_with_required_associations(event: str, response: Response, db: Session = Depends(get_db)):
```

Comment:
tutaj trzeba zdefiniować RequestModel (plik `models.py`) i podać go w endpoincie. Raczej nie używamy stringów do takich rzeczy

Reply:
@GrzesS0 nadal aktualne

Reply:
Nie bardzo rozumiem, co masz na myśli. To jest json zapisany jako string i potem jest parsowany do jsona. Nie bardzo wiem, jak inaczej mogło by to działać.

Reply:
Chodzi o to że jak robimy parametr jako string, to:
- nie ma możliwości walidacji pól
- nie wiemy jakie pola chcemy pobierać
- user może dać cokolwiek i nie mamy na to wpływu

Po to są modele jako paramerty, żeby nie dawać stringa i potem do z powrotem parsować.